Rought draft of  Napari x Xarray blog post with a roadmap

ianhi opened the pull request:
This is a follow up to scipy 2025 meeting. A public post of the plan we came up to let the community know where things stand.

I will also open a PR on xarray.dev with the same post in order crosspost for better visibility.


 cc @TimMonko in particular

ianhi commented on source/blog/napari-xarray.md:
```diff
@@ -0,0 +1,158 @@
+- Re-indexing dimensions (e.g. Fiji’s Stack to HyperStack) is difficult with unnamed dimensions  
+- Dimensionality reductions can lead to incorrect dimension mappings in the viewer due to dropping of singleton dimensions.
+  - For example, a max projected image stack will no longer respond to the correct sliders as the base image (`TCZYX` vs `TCYX`)
+- Users expect that provided dimension names will be matched to viewer dimension names, but presently adding a `TCZYX` image and `TYX` image will lead to the `TYX` image being aligned with `ZYX` in the viewer.
```
You're referring to the last one right? I really struggled with how to say this simply. maybe this is something where we need a visual example to show people.

jni reviewed (commented):
Not quite done with a full read through but I've left a few comments for now. Most important is that we *already* have axis_labels on layers and we should hook that up to the dims model!

TimMonko commented on source/blog/napari-xarray.md:
```diff
@@ -0,0 +1,163 @@
+That issue's age is a testament to the difficulty of the problems involved in this integration.
+
+But something being hard is not a reason not to try to do it.
+Indeed, one of the most successful plugins for Napari is [napari-aicsimageio](https://github.com/AllenCellModeling/napari-aicsimageio) which is built on top of [aicsimageio](https://github.com/AllenCellModeling/aicsimageio) (now superseded by [bioio](https://bioio-devs.github.io/bioio/OVERVIEW.html)), whose data model is backed by xarray.
```
haha that was me vs Ian. I used lower case and he used upper and I told him to pick one (our preference seems to be lower, when its stylistically appropriate)

psobolewskiPhD replied:
Well, napari should be canonically lower case. 
Other than the logo, Xarray docs appear to consistently use Xarray whether at the start of a sentence or not. So it should be Xarray here as well.

TimMonko commented:
> @TimMonko & co. I haven't done a full read-through but I wanted to say I do have a photo of y'all hacking on this from the sprints, I'm happy to send it through if we'd like to include it! Could be nice to get a bit of community feel in there

@draga that would be fab. Maybe DM us on zulip or send via email?

psobolewskiPhD commented on source/blog/napari-xarray.md:
```diff
@@ -0,0 +1,163 @@
+
+However, there are still several key pain points around managing image metadata when using Napari.
+
+- Users think in physical units (microns, lat/lon) rather than pixels
```
Most? 😉

psobolewskiPhD commented on source/blog/napari-xarray.md:
```diff
@@ -0,0 +1,163 @@
+- Re-indexing dimensions (e.g. Fiji’s Stack to HyperStack) is difficult with unnamed dimensions  
+- Dimensionality reductions can lead to incorrect dimension mappings in the viewer due to dropping of singleton dimensions.
+  - For example, a max projected image stack will no longer respond to the correct sliders as the base image (`TCZYX` vs `TCYX`)
+- Users expect that provided dimension names will be matched to viewer dimension names, but presently adding a `TCZYX` image and `TYX` image will lead to the `TYX` image being aligned with `ZYX` in the viewer.
```
```suggestion
- Users expect that provided dimension names will be matched to viewer dimension names, but napari right-aligns axes, so if you add an image with axes `TYX` to a viewer with a `TCZYX` image, then the axes of the `TYX` image will be aligned with the 3 right-most axes, so `ZYX`, in the viewer.
```

psobolewskiPhD commented on source/blog/napari-xarray.md:
```diff
@@ -0,0 +1,163 @@
+
+## How
+
+### SciPY 2025
```
```suggestion
### SciPy 2025
```

psobolewskiPhD commented on source/blog/napari-xarray.md:
```diff
@@ -0,0 +1,163 @@
+
+### SciPY 2025
+
+Thanks to generous support from the Chan Zuckerberg Institute the napari, Xarray, and [CellProfiler](https://cellprofiler.org/) team members were able to attend SciPy 2025. At the sprints members of these three teams—Ian Hunt-Isaak, Tim Monko, Nodar Gogoberidze, Peter Psobolewski and Carol Willing—collaborated to develop a plan for enhancing the integration of Xarray with Napari. The rest of this ...
```
Sorry for the lateness, but I do prefer my actual last name, Sobolewski.

psobolewskiPhD commented on source/blog/napari-xarray.md:
```suggestion
**Goal:** Handle a few clear cases of non-standard dimension ordering, not solve the general case of arbitrary reordering.
```

ianhi commented on source/blog/napari-xarray.md:
This is a much clearer and empathetic section with these edits

ianhi commented on source/blog/napari-xarray.md:
>   Not quite done with a full read through but I've left a few comments for now. Most important is that we already have axis_labels on layers and we should hook that up to the dims model!

in the spirit of finishing this, and brevity to keep the reader engaged I added this section that gets at what I htink @jni was suggesting without getting too caught up in detailed planning here.

willingc replied:
I like the framing of this. Let's go for it.

ianhi commented:
Hi all,

sorry for my delinquency here. I've had a go at addressing all the comments (less the TODO).

I have:

1. cleaned capitalization
2. Applied lots of suggested edits
3. Adding a video demo of the misaligned dim reduction
4. added a short section on `axis_labels`

Is there anything more we should be doing here? I see from @TimMonko's commment that I missed a PR review over on napari-xarray that should get done first, which I look at next.

 >  I'm happy to send it through if we'd like to include it! Could be nice to get a bit of community feel in there

I think I can find this in zulip. I'll look into adding it

ianhi commented on source/blog/napari-xarray.md:
```diff
@@ -0,0 +1,174 @@
+---
+
+# Xarray ❤️ napari
+
```
in addition to the date in the frontmatter?

TimMonko replied:
yes, otherwise it doesn't show up in the blog itself. And i think it would be nice to have a dated history of when these announcements are

willingc reviewed (approved):
This looks really good. Thanks @ianhi and team. Let's work on getting this merged soon. 🚀
